src/arithmetic.jl

```diff
@@ -208,28 +208,66 @@ end
 # Multiplication
 function *(a::TaylorModelN, b::TaylorModelN)
     @assert a.x0 == b.x0 && a.I == b.I
-
-    # Polynomial product with extended order
-    order = max(get_order(a), get_order(b))
-    @assert 2*order ≤ get_order()
-    aext = TaylorN(copy(a.pol.coeffs), 2*order)
-    bext = TaylorN(copy(b.pol.coeffs), 2*order)
-    res = aext * bext
-
-    # Returned polynomial
-    bext = TaylorN( copy(res.coeffs[1:order+1]) )
-
-    # Bound for the neglected part of the product of polynomials
-    res[0:order] .= zero(eltype(res))
     aux = a.I - a.x0
-    Δnegl = res(aux)
 
-    # Remainder of the product
-    Δa = a.pol(aux)
-    Δb = b.pol(aux)
-    Δ = Δnegl + Δb * a.rem + Δa * b.rem + a.rem * b.rem
+    # Some definitions related to the order of some polynomials
+    orderTS = get_order()  # maximum order, fixed by `TaylorSeries._params_TaylorN_`
+    order_a = get_order(a)
+    order_b = get_order(b)
+    order_prod = 2 * max(order_a, order_b)
+
+    # The returned polynomial has no terms beyond `orderTS`
+    if order_prod ≤ orderTS
+        apol = TaylorN(a.pol.coeffs, order_prod)
+        bpol = TaylorN(b.pol.coeffs, order_prod)
+        res = apol * bpol
+        Δa = a.pol(aux)
+        Δb = b.pol(aux)
+        Δ = Δb * a.rem + Δa * b.rem + a.rem * b.rem
+        return TaylorModelN(res, Δ, a.x0, a.I)
+    end
+
+    # The resulting product has a part whose order is larger than `orderTS`;
+    # a bound for this polynomial has to be included
+    apol = TaylorN(a.pol.coeffs, orderTS)
+    bpol = TaylorN(b.pol.coeffs, orderTS)
+    res = apol * bpol    # Trunctated polynomial to order `orderTS`
+    Δa = apol(aux)
+    Δb = bpol(aux)
+    Δ = Δb * a.rem + Δa * b.rem + a.rem * b.rem
+    N = get_numvars()
+
+    # We evaluate *explicitely* each term of the product of coefficients at `aux`
+    Δnegl = zero(Δ)
+    for order = orderTS+1:order_prod
+        orderini = order - orderTS
+        for inda = orderini:order_a
+            indb = order - inda
+            # Δnegl += evaluate(apol[inda], aux) * evaluate(bpol[indb], aux)
```

Comment:
You could try evaluating all of these bounds once before the loop (and storing them in new arrays), and then just indexing into the arrays inside the `for` loops. This should be much faster.

What I had done previously was store these bounds in the `TaylorModel` object itself when it is created (so that they are calculated exactly once in the history of the object).